integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/RddExecutionContext.java

```diff
@@ -248,7 +242,8 @@ public void end(SparkListenerJobEnd jobEnd) {
     eventEmitter.emit(event);
   }
 
-  protected OpenLineage.RunFacets buildRunFacets(ErrorFacet jobError, OpenLineage ol) {
+  protected OpenLineage.RunFacets buildRunFacets(
```

Comment:
I think this method should return RunFacetsBuilder instead of RunFacets. This would allow adding extra facet for JobStart event only and would simplify overall code of the PR.

Reply:
In the previous code, this facet reads event.properties while being built (I am not sure why because I think values in conf is enough, correct me if I was wrong).  When I modify this facet, I try my best to keep the previous logic. That's why the behavior of JobStart looks different from others. I wonder if we can stop collecting values from event.properties, or do you have any suggestions about this?